### PR TITLE
Fix the DCGAN example

### PR DESCRIPTION
**Pull request by wangkuiyi**

### CODE REVIEW

File: example/dcgan/README.md [hunk 1]
```diff
@@ -17,42 +17,50 @@ Training loss of PyTorch:
...
 ```bash
-cd $DATAROOT
 unzip img_align_celeba.zip
-find . | sort -R | tar czvf train.tgz -T -
+tar czf train.tgz img_align_celeba
 ```
 
-## Build
+NOTE: If you work on macOS, you must install GNU tar.  Go's standard package
+doesn't recognize macOS's BSD tar format.
```
Comment (wangkuiyi) [inline example/dcgan/README.md:36]: This note is very important to run this demo on macOS.

File: example/dcgan/dcgan.go [hunk 2]
```diff
@@ -3,9 +3,10 @@ package main
 import (
 	"flag"
 	"fmt"
-	"io"
+	_ "image/gif"
```
Comment (wangkuiyi) [inline example/dcgan/dcgan.go:6]: Without these anonymous imports, `image.Decode` cannot recognize JPEG file format.

File: example/dcgan/dcgan.go [hunk 3]
```diff
@@ -17,7 +18,7 @@ import (
 	"github.com/wangkuiyi/gotorch/vision/transforms"
 )
 
-var dataroot = flag.String("dataroot", "", "path to dataset")
+var data = flag.String("data", "", "path to dataset")
```
Comment (wangkuiyi) [inline example/dcgan/dcgan.go:21]: It is not a root directory, but a filename.

File: vision/datasets/imageloader.go [hunk 4]
```diff
@@ -44,15 +46,12 @@ func NewImageLoader(fn string, vocab map[string]int64, trans *transforms.Compose
...
-	if p.err == io.EOF {
-		return false
-	}
 	if p.err != nil {
 		return false
 	}
 	p.tensorGC()
 	p.retreiveMinibatch()
-	return true
+	return p.err == nil
```
Comment (wangkuiyi) [inline vision/datasets/imageloader.go:54]: This is a bug that prevents the loader from returning false when retrieeMInibatch fails.

File: example/dcgan/dcgan.go [hunk 5]
```diff
@@ -92,23 +93,20 @@ func discriminator(nc int64, ndf int64) *nn.SequentialModule {
...
-	loader, e := datasets.NewImageLoader(dataroot, vocab, trans, mbSize)
+	loader, e := datasets.NewImageLoader(data, vocab, trans, mbSize)
 	if e != nil {
 		panic(e)
 	}
 	return loader
 }
 
 func main() {
-	logFile, _ := os.OpenFile("gotorch-dcgan.log", os.O_CREATE|os.O_TRUNC|os.O_RDWR, 0666)
```
Comment (wangkuiyi) [inline example/dcgan/dcgan.go:108]: We can use bash script to do this job.

```bash
dcgan ... 2> >(tee dcgan.log)
```

File: example/dcgan/dcgan.go [hunk 6]
```diff
@@ -92,23 +93,20 @@ func discriminator(nc int64, ndf int64) *nn.SequentialModule {
 	)
 }
 
-func celebaLoader(dataroot string, vocab map[string]int64, mbSize int) *datasets.ImageLoader {
-	trans := transforms.Compose(transforms.Resize(64),
-		transforms.CenterCrop(64),
```
Comment (wangkuiyi) [inline example/dcgan/dcgan.go:97]: 64 here is a magic number.

**Review by QiJune — APPROVED** [review]
LGTM

**Pull request merged**